spatialmedia/spherical.py

```diff
@@ -0,0 +1,377 @@
+    "<GSpherical:FullPanoHeightPixels>{3}</GSpherical:FullPanoHeightPixels>"\
+    "<GSpherical:CroppedAreaLeftPixels>{4}</GSpherical:CroppedAreaLeftPixels>"\
+    "<GSpherical:CroppedAreaTopPixels>{5}</GSpherical:CroppedAreaTopPixels>"
+
+SPHERICAL_XML_FOOTER = "</rdf:SphericalVideo>"
+
+SPHERICAL_TAGS_LIST = [
+    "Spherical",
+    "Stitched",
+    "StitchingSoftware",
+    "ProjectionType",
+    "SourceCount",
+    "StereoMode",
+    "InitialViewHeadingDegrees",
+    "InitialViewPitchDegrees",
+    "InitialViewRollDegrees",
+    "Timestamp",
+    "CroppedAreaImageWidthPixels",
+    "CroppedAreaImageHeightPixels",
+    "FullPanoWidthPixels",
+    "FullPanoHeightPixels",
+    "CroppedAreaLeftPixels",
+    "CroppedAreaTopPixels",
+]
+
+SPHERICAL_PREFIX = "{http://ns.google.com/videos/1.0/spherical/}"
+SPHERICAL_TAGS = dict()
+for tag in SPHERICAL_TAGS_LIST:
+    SPHERICAL_TAGS[SPHERICAL_PREFIX + tag] = tag
+
+integer_regex_group = "(\d+)"
+crop_regex = "^{0}$".format(":".join([integer_regex_group] * 6))
+
+
+def spherical_uuid(metadata):
+    """Constructs a uuid containing spherical metadata.
+
+    Args:
+      metadata: String, xml to inject in spherical tag.
+
+    Returns:
+      uuid_leaf: a box containing spherical metadata.
+    """
+    uuid_leaf = mpeg.Box()
+    assert(len(SPHERICAL_UUID_ID) == 16)
+    uuid_leaf.name = mpeg.constants.TAG_UUID
+    uuid_leaf.header_size = 8
+    uuid_leaf.content_size = 0
+
+    uuid_leaf.contents = SPHERICAL_UUID_ID + metadata
+    uuid_leaf.content_size = len(uuid_leaf.contents)
+
+    return uuid_leaf
+
+
+def mpeg4_add_spherical(mpeg4_file, in_fh, metadata):
+    """Adds a spherical uuid box to an mpeg4 file for all video tracks.
+
+    Args:
+      mpeg4_file: mpeg4, Mpeg4 file structure to add metadata.
+      in_fh: file handle, Source for uncached file contents.
+      metadata: string, xml metadata to inject into spherical tag.
+    """
+    for element in mpeg4_file.moov_box.contents:
+        if element.name == mpeg.constants.TAG_TRAK:
+            added = False
+            element.remove(mpeg.constants.TAG_UUID)
+            for sub_element in element.contents:
+                if sub_element.name != mpeg.constants.TAG_MDIA:
+                    continue
+                for mdia_sub_element in sub_element.contents:
+                    if mdia_sub_element.name != mpeg.constants.TAG_HDLR:
+                        continue
+                    position = mdia_sub_element.content_start() + 8
+                    in_fh.seek(position)
+                    if in_fh.read(4) == mpeg.constants.TRAK_TYPE_VIDE:
+                        added = True
+                        break
+
+                if added:
+                    if not element.add(spherical_uuid(metadata)):
+                        return False
+                    break
+
+    mpeg4_file.resize()
+    return True
+
+
+def ParseSphericalXML(contents, console):
```

Comment:
function naming should be consistent within this file and package: use small_case. If you're trying to distinguish between public/private, a leading underscore is allowable in pep-8/google styles.